### PR TITLE
Add support for new silence matcher types

### PR DESCRIPTION
**Pull request by QuentinBisson**
We can actually keep the existing silence CRD by adding a new value set to true by default

## Checklist

- [x] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by QuentinBisson** [comment]
I added the new field from https://github.com/prometheus/alertmanager/blob/c72c4d79f6d94429057944820b1c08767d540134/pkg/labels/matcher.go#L112

**Comment by mig4** [comment]
A couple of questions, maybe daft but I feel like I'm missing some context:

- what's the goal exactly? the title says _support for new silence matcher types_ but in the diff I can only see a single field added; what are the new matcher types, etc.? Any chance you could link up to some references so I can catch up?
- in PR description you say _adding a new value false by default_ but in the code `IsEqual` is set to `true`, is this right?

**Comment by QuentinBisson** [comment]
Sorry for the confusion, the value should be set to true by default.

Regarding the new field, the alertmanager folks added it to keep the api compatibility between v1 and v2 of the silence api.

They added the IsEqual field https://github.com/prometheus/alertmanager/blob/c72c4d79f6d94429057944820b1c08767d540134/pkg/labels/matcher.go#L95
This is set to true by default because the old api only supported equal matcher types so `=` or `~=`. 
As you can see here, the new Matcher type is the internal representation but the http object is still the old silence type:
https://github.com/prometheus/alertmanager/blob/c72c4d79f6d94429057944820b1c08767d540134/pkg/labels/matcher.go#L121

Basically, adding this field with a default value of true will not change the current behavior but we can now support negative regexes by setting isEqual to false in the Silence CRs

**Comment by mig4** [comment]
> They added the IsEqual field [prometheus/alertmanager@`c72c4d7`/pkg/labels/matcher.go#L95](https://github.com/prometheus/alertmanager/blob/c72c4d79f6d94429057944820b1c08767d540134/pkg/labels/matcher.go#L95)
> This is set to true by default because the old api only supported equal matcher types so `=` or `~=`.
> As you can see here, the new Matcher type is the internal representation but the http object is still the old silence type:
> [prometheus/alertmanager@`c72c4d7`/pkg/labels/matcher.go#L121](https://github.com/prometheus/alertmanager/blob/c72c4d79f6d94429057944820b1c08767d540134/pkg/labels/matcher.go#L121)

Ah, got ya, so `IsEqual = false` means match not equal to, i.e. `!=` or `!~`. Nice, thanks for explaining 😁

**Review by mig4 — APPROVED** [review]
